rero_ils/modules/documents/serializers.py

```diff
@@ -30,6 +30,92 @@
 from ..organisations.api import Organisation
 from ..serializers import JSONSerializer, RecordSchemaJSONV1
 
+DOCUMENT_TYPES = {
+    'docmaintype_archives': {},
+    'docmaintype_article': {},
+    'docmaintype_audio': {
+        'docsubtype_music': True,
+        'docsubtype_sound': True,
+        'docsubtype_audio_book': True,
+        'docsubtype_recorded_words': True
+    },
+    'docmaintype_book': {
+        'docsubtype_other_book': True,
+        'docsubtype_large_print_book': True,
+        'docsubtype_braille_book': True,
+        'docsubtype_manuscript': True,
+        'docsubtype_thesis': True,
+        'docsubtype_e-book': True,
+        'docsubtype_microform': True
+    },
+    'docmaintype_children': {},
+    'docmaintype_comic': {
+        'docsubtype_comic': True,
+        'docsubtype_manga': True
+    },
+    'docmaintype_electronic': {
+        'docsubtype_cdrom_dvdrom': True,
+        'docsubtype_other_electronic_carrier': True
+    },
+    'docmaintype_game': {
+        'docsubtype_other_game': True,
+        'docsubtype_video_game': True,
+        'docsubtype_role-playing_game': True,
+        'docsubtype_board_game': True
+    },
+    'docmaintype_image': {
+        'docsubtype_poster': True,
+        'docsubtype_postcard': True,
+        'docsubtype_photography': True,
+        'docsubtype_print_engraving': True,
+        'docsubtype_calendar': True,
+        'docsubtype_other_image': True
+    },
+    'docmaintype_language_method': {},
+    'docmaintype_leaf': {},
+    'docmaintype_map': {
+        'docsubtype_map': True,
+        'docsubtype_atlas': True
+    },
+    'docmaintype_movie_series': {
+        'docsubtype_movie': True,
+        'docsubtype_series': True,
+        'docsubtype_documentary': True
+    },
+    'docmaintype_object': {
+        'docsubtype_kamishibai': True,
+        'docsubtype_tablet': True,
+        'docsubtype_other_object': True
+    },
+    'docmaintype_other': {},
+    'docmaintype_score': {
+        'docsubtype_printed_score': True,
+        'docsubtype_music_method': True,
+        'docsubtype_manuscript_score': True
+    },
+    'docmaintype_serial': {},
+    'docmaintype_series': {}
+}
+
+
+def filter_document_type_buckets(buckets):
+    """Removes unwanted sub types from buckets."""
+    new_type_buckets = []
+    for type_bucket in buckets:
+        new_type_bucket = type_bucket
+        main_type = type_bucket['key']
+        new_subtype_buckets = []
+        subtype_buckets = type_bucket['document_subtype']['buckets']
+        for subtype_bucket in subtype_buckets:
+            if DOCUMENT_TYPES.get(main_type, {}).get(subtype_bucket['key']):
+                new_subtype_buckets.append(subtype_bucket)
+        new_type_bucket[
+            'document_subtype'
+        ]['buckets'] = new_subtype_buckets
+        print(main_type, new_subtype_buckets)
+        new_type_buckets.append(new_type_bucket)
+    return new_type_buckets
+
```

Comment:
Maybe interesting to think how to "remove" unwanted key from `buckets` arguments instead of recreate a new complete buckets ? In this case the L110-L111 should be something like 
```
for subtype_idx, subtype_bucket in enumerate(subtype_buckets):
    if subtype_bucket['key'] not in DOCUMENT_TYPES.get(main_type, {}):
        del buckets[bucket_idx]['document_subtype']['buckets'][subtype_idx]
```
I'm not sure about the bucket structure, you know it better than me.

For me it will increase the function readability.